azext_edge/tests/edge/orchestration/resources/connector/opcua/test_opcua_certs_unit.py

```diff
@@ -0,0 +1,913 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import os
+from typing import List, Optional
+from unittest.mock import Mock
+import pytest
+
+import responses
+from azext_edge.edge.commands_connector import (
+    add_connector_opcua_client,
+    add_connector_opcua_issuer,
+    add_connector_opcua_trust,
+)
+from azext_edge.edge.providers.orchestration.resource_map import IoTOperationsResource
+from azext_edge.edge.providers.orchestration.resources.connector.opcua.certs import (
+    OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+    OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+    OPCUA_SPC_NAME,
+    OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+)
+from azext_edge.edge.providers.orchestration.work import IOT_OPS_EXTENSION_TYPE
+from azext_edge.tests.edge.orchestration.resources.conftest import get_base_endpoint, get_mock_resource
+from azext_edge.tests.generators import generate_random_string
+
+
+@pytest.fixture
+def mocked_logger(mocker):
+    patched = mocker.patch(
+        "azext_edge.edge.providers.orchestration.resources.connector.opcua.certs.logger",
+    )
+    yield patched
+
+
+@pytest.fixture
+def mocked_get_resource_client(mocker):
+    patched = mocker.patch("azext_edge.edge.util.queryable")
+    yield patched().get_resource_client
+
+
+@pytest.fixture
+def mocked_resource_map(mocker):
+    patched = mocker.patch(
+        "azext_edge.edge.providers.orchestration.resources.connector.opcua.certs.Instances",
+    )
+    yield patched().get_resource_map
+
+
+@pytest.fixture
+def mocked_sleep(mocker):
```

Comment:
you can move anything that isnt specific to opcua (as in the file path does not have opcua) to conftest